### PR TITLE
AuTest: bind stdout/stderr to traffic.out

### PR DESCRIPTION
**Pull request by bneradt**
This updates our AuTest invocation of traffic_server to bind its stdout
and stderr output to traffic.out. Doing this allows the tests to
dynamically interact with the traffic.out content, such as waiting upon
certian debug tag lines. It also serves as a debugging convenience since
all logging output for a Traffic Server process will be in its log dir:
diags.log, error.log, etc., and now traffic.out.

### CODE REVIEW

File: tests/gold_tests/pluginTest/lua/lua_states_stats.test.py [hunk 1]
```diff
@@ -28,12 +28,7 @@
 # Define default ATS
 server = Test.MakeOriginServer("server")
 
-# It is necessary to redirect stderr to a file so it will be available for examination by a test run.
-ts = Test.MakeATSProcess(
-    "ts", command="traffic_manager 2> " + Test.RunDirectory + "/ts.stderr.txt", select_ports=True
-)
```
Comment (bneradt) [inline tests/gold_tests/pluginTest/lua/lua_states_stats.test.py:34]: This is an example of something no longer necessary if we bind stdout and stderr to traffic.out. Instead we can simply interact with traffic.out.

**Review by randall — APPROVED** [review]

**Comment by marcin-wadolkowski** [comment]
I'm not sure if it's an issue. I noticed that in file:
tests/gold_tests/pluginTest/stek_share/stek_share.test.py

Lines 225 to 229 begin with **ts1**.Disk.traffic_out.Content but maybe should begin with:
**ts1**.Disk.traffic_out.Content (...)
**ts2**.Disk.traffic_out.Content (...) 
...
**ts5**.Disk.traffic_out.Content (...)

**Comment by bneradt** [comment]
> I'm not sure if it's an issue. I noticed that in file: tests/gold_tests/pluginTest/stek_share/stek_share.test.py
> 
> Lines 225 to 229 begin with **ts1**.Disk.traffic_out.Content but maybe should begin with: **ts1**.Disk.traffic_out.Content (...) **ts2**.Disk.traffic_out.Content (...) ... **ts5**.Disk.traffic_out.Content (...)

Good catch, @marcin-wadolkowski . Fixing in #8921.

**Comment by zwoop** [comment]
We need this for other PRs to CP.

Cherry-picked to v9.2.x

**Pull request merged**